module/src/main/java/org/mule/extension/spring/internal/beanfactory/ArtifactObjectsAwareBeanFactory.java

```diff
@@ -105,22 +95,22 @@ public Object doResolveDependency(DependencyDescriptor descriptor, String beanNa
    */
   @Override
   protected <T> T doGetBean(String name, Class<T> requiredType, Object[] args, boolean typeCheckOnly) throws BeansException {
-    if ("fips140-2".equals(System.getProperty("mule.security.model"))) {
-      if (name.contains(DaoAuthenticationProvider.class.getName()) || DaoAuthenticationProvider.class.equals(requiredType)) {
-        return (T) authenticationProvider(getUserDetailService(name));
-      }
+    if ("fips140-2".equals(System.getProperty("mule.security.model"))
+        && (name.contains(DaoAuthenticationProvider.class.getName()) || DaoAuthenticationProvider.class.equals(requiredType))) {
+      return (T) authenticationProvider(getUserDetailService(name));
+
     }
     if (containsBean(name) || !artifactObjectProvider.containsObject(name) || destroying) {
       return super.doGetBean(name, requiredType, args, typeCheckOnly);
     } else {
-      return (T) artifactObjectProvider.getObject(name).get();
+      return (T) artifactObjectProvider.getObject(name).get(); // NOSONAR
```

Comment:
Is this related to the other `NOSONAR` we discussed before? or is it a different rule?